File: contest/atcoder/s8pc-2/a/0.cpp
// 15 と 36 で WA。
#include <cstdio>

int main(void)
{
	int required = 1;
	int answer = 0;
	while (true) {
		char c;
		scanf("%c", &c);
		int input;
		if (c == 'I') {
			input = 1;
		} else if (c == 'O') {
			input = 0;
		} else {
			break;
		}
		if (input == required) {
			answer++;
			required = (required + 1) % 2;
		}
	}
	if (answer % 2 == 0) {
		answer--;
	}
	printf("%d\n", answer);
	return 0;
}

File: contest/atcoder/arc/arc053-b/0.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
from collections import Counter
string = input()
counter = dict(Counter(string))
num_pairs = 0
num_no_pairs = 0
for key in counter:
    num_pairs += counter[key] // 2
    num_no_pairs += counter[key] % 2
answer = 0
if num_no_pairs >= 1:
    answer = 2 * (num_pairs // num_no_pairs) + 1
else:
    answer = 2 * num_pairs
print(answer)
